monkestation/code/modules/station_goals/meteor_shield.dm

```diff
@@ -0,0 +1,105 @@
+GLOBAL_LIST_EMPTY(meteor_shield_sats)
+
+/obj/machinery/satellite/meteor_shield
+	/// Whether the meteor sat checks for line of sight to determine if it can intercept a meteor.
+	var/check_sight = TRUE
+	/// The proximity monitor used to detect meteors entering the shield's range.
+	var/datum/proximity_monitor/meteor_monitor
+
+/obj/machinery/satellite/meteor_shield/Initialize(mapload)
+	. = ..()
+	GLOB.meteor_shield_sats += src
+	setup_proximity()
+
+/obj/machinery/satellite/meteor_shield/Destroy()
+	GLOB.meteor_shield_sats -= src
+	if(meteor_monitor)
+		QDEL_NULL(meteor_monitor)
+	return ..()
+
+/obj/machinery/satellite/meteor_shield/vv_edit_var(vname, vval)
+	switch(vname)
+		if(NAMEOF(src, kill_range))
+			meteor_monitor?.set_range(kill_range)
+			kill_range = vval
+			datum_flags |= DF_VAR_EDITED
+			return TRUE
+		if(NAMEOF(src, active))
+			active = vval
+			datum_flags |= DF_VAR_EDITED
+			setup_proximity()
+			return TRUE
+	return ..()
+
+/obj/machinery/satellite/meteor_shield/HasProximity(obj/effect/meteor/meteor)
+	if(!active || !istype(meteor) || QDELING(meteor) || (obj_flags & EMAGGED))
+		return
+	var/turf/our_turf = get_turf(src)
+	var/turf/meteor_turf = get_turf(meteor)
+	if(!check_los(our_turf, meteor_turf))
+		return
+	our_turf.Beam(meteor_turf, icon_state = "sat_beam", time = 5)
+	if(meteor.shield_defense(src))
+		new /obj/effect/temp_visual/explosion(meteor_turf)
+		INVOKE_ASYNC(src, PROC_REF(play_zap_sound), meteor_turf)
+		qdel(meteor)
+
+/obj/machinery/satellite/meteor_shield/proc/check_los(turf/source, turf/target) as num
+	// if something goes fucky wucky, let's just assume line-of-sight by default
+	if(!check_sight)
+		return TRUE
+	for(var/turf/segment as anything in get_line(source, target))
```

Comment:
I swear there's a proc for getting a straight line w/o obstruction already

Reply:
if there was i'd have used it when i was doing adaptive jps
(psst, there isnt)